platforms/compare_test.go

```diff
@@ -75,6 +81,18 @@ func TestOnly(t *testing.T) {
 				},
 			},
 		},
+		{
+			platform: fmt.Sprintf("windows/amd64//%s", buildStr+".1"),
```

Comment:
Slightly orthogonal, but wondering if the string representation for osversion (and features) was already "formalised" . I recall there were some discussion about this in the past (but don't recall the outcome). We should probably have some of this added to the OCI specs at some point

Reply:
I wasn't sure if this valid string.  The parser doesn't actually work in this case but was trying to demonstrate this isn't really handled.

Reply:
Makes sense! Yes, I also was looking at Parse() not yet supporting it (and left that for a later exercise).

Empty `//` may work for missing components (but could potentially be problematic if someone (eg) uses `path.Join()` to construct the string).

Perhaps there's alternatives (`/-/` for explicitly omitting a field, or `/*/`); in any case, likely something for further discussion

Reply:
Discussing with someone else, and an alternative for the format could be `os/arch/variant@os-version[+feature+feature...]`

(we should look at the `+` symbol though, in case  it may be used in URLs (where `+` can be ambiguous)